lib/common/multithread.c

```diff
@@ -128,6 +128,7 @@ void h2o_multithread_destroy_queue(h2o_multithread_queue_t *queue)
     h2o_socket_read_stop(queue->async.read);
     h2o_socket_close(queue->async.read);
     close(queue->async.write);
+    free(queue->async.read);
```

Comment:
Memory allocated for `h2o_socket_t *` must be returned by one of the event loop handlers when their states get updated (e.g. at https://github.com/h2o/h2o/blob/5e828100f261bc9952cbf8031fbfb1ef05286591/lib/common/socket/evloop/epoll.c.h#L47).

If you are seeing a leak for `queue->async.read`, then I believe it should be fixed by making the application run the event loop once more.